app/routes.py

```diff
@@ -1 +1,272 @@
+        task.title = request_body["title"]
+        task.description = request_body["description"]
+    except KeyError:
+        return {
+            "msg": "Update failed due to missing data. Title, Description are required!"
+        }, 400
+
+    db.session.commit()
+
+    rsp = {"task" : task.get_dict()}
+    return jsonify(rsp), 200
+
+
+@tasks_bp.route("/<task_id>", methods=["DELETE"])
+def delete_one_task(task_id):
+    task = validate_task(task_id)
+
+    db.session.delete(task)
+    db.session.commit()
+
+    rsp = {"details": f'Task {task_id} "{task.title}" successfully deleted'}
+    return jsonify(rsp), 200
+
+@tasks_bp.route("/<task_id>/mark_complete", methods=["PATCH"])
+def mark_task_complete(task_id):
+    task = validate_task(task_id)
+    task.completed_at = datetime.now()
+    db.session.commit()
+    
+    Slack_Url = 'https://slack.com/api/chat.postMessage'
+    message = {"text": f"Someone just completed the task {task.title}", "channel": "C049FQLJTBN"}
+    token = os.environ.get('SLACKBOT-TOKEN')
+    requests.post(Slack_Url, json=message, headers={"Authorization": f"Bearer {token}"})
+
+    rsp = {"task": task.get_dict()}
+    return jsonify(rsp), 200
+
+
+@tasks_bp.route("/<task_id>/mark_incomplete", methods=["PATCH"])
+def mark_task_incomplete(task_id):
+    task = validate_task(task_id)
+    task.completed_at = None
+
+    db.session.commit()
+
+    rsp = {"task": task.get_dict()}
+    return jsonify(rsp), 200
+
+
+def validate_task(task_id):
+    try:
+        task_id = int(task_id)
+    except:
+        rsp = {"msg": f"Task with id #{task_id} is invalid."}
+        abort(make_response(rsp, 400))
+    
+    task = Task.query.get(task_id)
+
+    if not task:
+        rsp = {"msg": f"Task with id #{task_id} is not found!"}
+        abort(make_response(rsp, 404))
+    return task
+
+
+# Goal routes
+@goals_bp.route("", methods=["POST"])
+def create_one_goal():
+    if not request.is_json:
+        return {"msg": "Missing JSON request body"}, 400
+
+    request_body = request.get_json()
+    try:
+        title = request_body["title"]
+
+    except KeyError:
+        return {"details": "Invalid data"}, 400
+
+    new_goal = Goal(title=title)
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    rsp = {"goal": new_goal.get_dict()}
+
+    return jsonify(rsp), 201
+
+
+@goals_bp.route("", methods=["GET"])
+def get_all_goals():
+    sort_query = request.args.get("sort")
+    goals = Goal.query.all()
+
+    if sort_query == "asc":
+        goals = Goal.query.order_by(sqlalchemy.asc(Goal.title))
+    elif sort_query == "desc":
+        goals = Goal.query.order_by(sqlalchemy.desc(Goal.title))
+
+    goals_response = []
+    for goal in goals:
+        goals_response.append(goal.get_dict())
+
+    return jsonify(goals_response), 200
+
+
+@goals_bp.route("/<goal_id>", methods=["GET"])
+def get_one_goal(goal_id):
+    goal = validate_goal(goal_id)
+    rsp = {"goal": goal.get_dict()}
+
+    return jsonify(rsp), 200
```

Comment:
`rsp` is already a dictionary so you don't need to `jsonify` it. So instead we can return:

```python
return rsp, 200
```

Reply:
Because we're returning a dictionary, we can take advantage of Flask's default behavior of turning dictionaries into JSON format as according to their documentation: 

> If you return a dict from a view, it will be converted to a JSON response. 

https://flask.palletsprojects.com/en/1.1.x/quickstart/#apis-with-json

We can apply this same comment to the rest of this project.